verify/verifier.go

```diff
@@ -66,31 +89,44 @@ func NewVerifier(cfg *Config, l log.Logger) (*Verifier, error) {
 	}
 
 	return &Verifier{
-		kzgVerifier: kzgVerifier,
-		verifyCerts: cfg.VerifyCerts,
-		cv:          cv,
+		log:                       log,
+		kzgVerifier:               kzgVerifier,
+		cv:                        cv,
+		rollupBlobInclusionWindow: cfg.RollupBlobInclusionWindow,
 	}, nil
 }
 
 // verifies V0 eigenda certificate type
-func (v *Verifier) VerifyCert(ctx context.Context, cert *Certificate) error {
-	if !v.verifyCerts {
-		return nil
-	}
-
+func (v *Verifier) VerifyCert(ctx context.Context, cert *Certificate, args common.VerifyArgs) error {
 	// 1 - verify batch in the cert is confirmed onchain
 	err := v.cv.verifyBatchConfirmedOnChain(ctx, cert.Proof().GetBatchId(), cert.Proof().GetBatchMetadata())
 	if err != nil {
 		return fmt.Errorf("failed to verify batch: %w", err)
 	}
 
-	// 2 - verify merkle inclusion proof
+	// 2 - verify that the blob cert was submitted to the rollup's batch inbox within the allowed RollupBlobInclusionWindow window.
+	// This is to prevent timing attacks where a rollup batcher could try to game the fraud proof window by including an old DA blob that is about to expire on the DA layer
+	// and is hence not retrievable.
+	if args.RollupL1InclusionBlockNum >= 0 && v.rollupBlobInclusionWindow > 0 {
+		blockNumEigenDA := int64(cert.BlobVerificationProof.BatchMetadata.BatchHeader.ReferenceBlockNumber)
+		blockNumRollup := args.RollupL1InclusionBlockNum
+		// We need blockNumEigenDA < blockNumRollup < blockNumEigenDA + rollupBlobInclusionWindow
+		if !(blockNumEigenDA < blockNumRollup) {
+			return fmt.Errorf("eigenda batch reference block number (%d) needs to be < rollup inclusion block number (%d)", blockNumEigenDA, blockNumRollup)
```

Comment:
Suggest adding a test case which covers this error, since the conditional is non trivial

Reply:
[47ff9ba](https://github.com/Layr-Labs/eigenda-proxy/pull/226/commits/47ff9ba71a9047b80f7b8af7d3f702d0fb0c97f8)